deterministic.c

```diff
@@ -5,14 +5,20 @@
 #include "inner.h"
 #include "deterministic.h"
 
+#define FALCON_DET1024_TMPSIZE_KEYGEN FALCON_TMPSIZE_KEYGEN(FALCON_DET1024_LOGN)
+#define FALCON_DET1024_TMPSIZE_SIGNDYN FALCON_TMPSIZE_SIGNDYN(FALCON_DET1024_LOGN)
+#define FALCON_DET1024_TMPSIZE_VERIFY FALCON_TMPSIZE_VERIFY(FALCON_DET1024_LOGN) 
+#define FALCON_DET1024_SALTED_SIG_COMPRESSED_MAXSIZE FALCON_SIG_COMPRESSED_MAXSIZE(FALCON_DET1024_LOGN)
+#define FALCON_DET1024_SALTED_SIG_CT_SIZE FALCON_SIG_CT_SIZE(FALCON_DET1024_LOGN)
+
+
 int falcon_det1024_keygen(shake256_context *rng, void *privkey, void *pubkey) {
-	size_t tmpkg_len = FALCON_TMPSIZE_KEYGEN(FALCON_DET1024_LOGN);
```

Comment:
I'm not familiar enough to know if it is a performance improvement to store the result of the expression inside the macro in a variable for use multiple times — WDYT? It seems like with the conditional expression it might be faster to only do it once?
```
#define FALCON_TMPSIZE_KEYGEN(logn) \
        (((logn) <= 3 ? 272u : (28u << (logn))) + (3u << (logn)) + 7)
```

Reply:
The point is not performance. you get a warning from the compiler on that. it is not recommended in C.

Reply:
so what's wrong with assigning the macro's expression to a size_t and using it multiple times in the function like the code was doing before ...? I think I am missing the point.

Reply:
Those lines of code:
`	size_t tmpkg_len = FALCON_TMPSIZE_KEYGEN(FALCON_DET1024_LOGN);
	uint8_t tmpkg[tmpkg_len];
`
produce this warning
`Warning: variable length array used [-Wvla]`

it is not recommending

Reply:
ohhh variable length array! I wasn't getting that from typing make on the compressed-ct branch on my Mac. OK I see.